shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1331,7 +1311,8 @@ - (void)createTouchRateCorrectionVSyncClientIfNeeded {
     return;
   }
 
-  flutter::Shell& shell = [_engine.get() shell];
+  NSAssert(self.engine, @"engine must not be nil");
```

Comment:
Might be safer to do your suggestion in https://github.com/flutter/flutter/issues/153971 and just return, since we suspect this case is happening?

Or add the `NSAssert`s in a follow-up commit so if it's a problem it doesn't result in a revert of the rest of the ARC changes in this PR?

Reply:
I'll do this in a followup commit. You're right, these are totally orthogonal.